### ISSUE TITLE
[Bug]: LNbank throws error on invoice creation

### ISSUE THREAD

**Issue by kaloudis** [btcpayserver/btcpayserver#4490]
### What is your BTCPay version?

BTCPay v1.7.3.0
LNbank 1.3.8.0

### How did you deploy BTCPay Server?

Extended manual setup. Using external LND.

### What happened?

LNbank throws error on invoice creation

### How did you encounter this bug?

When creating an invoice in LNbank, both through the BTCPay Server web app and through Zeus, I hit the following issue:

`Invoice creation failed: Exception of type 'BTCPayServer.JsonHttpException' was thrown.`

Revelant log output included.

### Relevant log output

```shell
Dec 24 23:46:07 45-79-192-236 run.sh[12490]: System.OverflowException: Value was either too large or too small for an Int64.
Dec 24 23:46:07 45-79-192-236 run.sh[12490]:    at System.Decimal.ToInt64(Decimal d)
Dec 24 23:46:07 45-79-192-236 run.sh[12490]:    at NBitcoin.Money.Coins(Decimal coins)
Dec 24 23:46:07 45-79-192-236 run.sh[12490]:    at BTCPayServer.Services.Invoices.PaymentMethod.Calculate(Func`2 paymentPredicate)
Dec 24 23:46:07 45-79-192-236 run.sh[12490]:    at BTCPayServer.HostedServices.InvoiceWatcher.GetNearestClearedPayment(PaymentMethodDictionary allPaymentMethods, PaymentMethodAccounting& accounting) in /home/bitcoin/src/btcpayserver/BTCPayServer/HostedServices/InvoiceWatcher.cs:line 213
Dec 24 23:46:07 45-79-192-236 run.sh[12490]:    at BTCPayServer.HostedServices.InvoiceWatcher.UpdateInvoice(UpdateInvoiceContext context)
Dec 24 23:46:07 45-79-192-236 run.sh[12490]:    at BTCPayServer.HostedServices.InvoiceWatcher.StartLoop(CancellationToken cancellation) in /home/bitcoin/src/btcpayserver/BTCPayServer/HostedServices/InvoiceWatcher.cs:line 320
```

### Are you sure this is a bug report?

- [X] I confirm this is a bug report

**Comment by iBobik** [comment]
Btw, there is workaround to switch to internal node for now.

**Comment by dennisreimann** [comment]
I still cannot reproduce it :( 

What LND version are you using and what does "external LND" mean? Does you BTCPay even have an internal Lightning node or are you configuring the store's Lightning node via connection string?

**Comment by kaloudis** [comment]
I'm basically using this guide https://docs.btcpayserver.org/Deployment/ManualDeploymentExtended/

I don't have the internal LN node enabled. I'm using a connection string.

**Comment by NicolasDorier** [comment]
@kaloudis which version of LND?

I am not sure this stack trace is related to `Invoice creation failed: Exception of type 'BTCPayServer.JsonHttpException' was thrown.`

You probably have another exception?

**Comment by iBobik** [comment]
I use officiall Docker stack with internal LND and version in that stack.

**Comment by dennisreimann** [comment]
@iBobik do you have the same problem as Evan?

**Comment by iBobik** [comment]
Yes. I just do not see System.OverflowException exception in the log, but I probably does not have debug enabled or something like this.

**Comment by dennisreimann** [comment]
@kaloudis By now I assume the `BTCPAY_BTCLIGHTNING` env variable isn't set in your manual deployment, because the docs didn't mention that. See the end of the [Lightning section](https://docs.btcpayserver.org/Deployment/ManualDeploymentExtended/#lightning-network-daemon-lnd) which I just added: Without this, the node isn't interpreted as the "internal node" — LNbank requires that, but doesn't check for its existence yet, I'll add that with the next version.

TL;DR Set you connection string to the env variable like this:

```bash
export BTCPAY_BTCLIGHTNING="type=lnd-rest;server=https://127.0.0.1:8080/;macaroonfilepath=/home/admin/.lnd/data/chain/bitcoin/mainnet/admin.macaroon;allowinsecure=true"
```

**Comment by kaloudis** [comment]
No dice. Hitting the same error.

**Comment by dennisreimann** [comment]
Ok, then we need to see if the upcoming versions change something in that regard.

As said, I haven't been able to reproduce it and I'm not optimistic that the update will fix it.

**Comment by dennisreimann** [comment]
@kaloudis What are the amount and currency of the invoice, as well as the rate source for the store?

**Comment by kaloudis** [comment]
> @kaloudis What are the amount and currency of the invoice, as well as the rate source for the store?

variable amounts, SATS, and coingecko

**Comment by kaloudis** [comment]
Probablem persists with BTCPay Server v1.7.4  and Lnbank 1.3.10

**Comment by dennisreimann** [comment]
@kaloudis Thanks for reporting, we'll solve this one day ;)

> though I now see the LNbank requires an internal Lightning node to be configured. messsage

@NicolasDorier Evan ist using the manual deployment method, but added the `BTCPAY_BTCLIGHTNING` env variable as described above. Do you have an idea why he still gets this message? 

Maybe it needs the `. ./btcpay-setup.sh -i` after setting it? Not sure if you ran this.

Just checked and for the Docker deployment it works reliably, so I'm pretty sure that setting didn't get picked up somehow.

**Comment by NicolasDorier** [comment]
If you did something like
```
BTCPAY_BTCLIGHTNING: "type=lnd-rest;server=http://lnd_bitcoin:8080/;macaroonfilepath=/etc/lnd_bitcoin/admin.macaroon;allowinsecure=true"
```

Then can you check you can create a new store as admin with this internal node? if yes, then this is correctly configured.

**Comment by kaloudis** [comment]
I am not running the dockerized version but my env var is set.

I was able to create a new admin with this internal node but had to manually paste in the connection string

I am still seeing that error regarding an internal node on the LNbank screen

I can however, still delete and create LNbank accounts

I still hit the same error as reported above when trying to create an invoice with a LNbank account

**Comment by dennisreimann** [comment]
@NicolasDorier It looks like the manual deployment doesn't pick up the Lightning connection string via the env var — not sure why, but I saw that [the guide recommends a config file](https://docs.btcpayserver.org/Deployment/ManualDeploymentExtended/#3-create-a-configuration-file), so those might conflict.

@kaloudis Do you have the `btcpay.config` from the guide? If so, can you check if setting this makes a difference?

```
BTC.lightning=type=lnd-rest;server=http://lnd_bitcoin:8080/;macaroonfilepath=/etc/lnd_bitcoin/admin.macaroon;allowinsecure=true
```

If so, I'll adapt the docs to set it in the config file instead of using the environment variable.

Sorry for the hassle, but I don't have a manual deployment to test this with.

**Comment by kaloudis** [comment]
@dennisreimann you finally cracked it!
